libraries/Standard/src/math/special.birch

```diff
@@ -90,36 +104,79 @@ function lgamma(x:Real32, p:Integer) -> Real32 {
  * The beta function.
  */
 function beta(x:Real64, y:Real64) -> Real64 {
-  cpp {{
-  return boost::math::beta(x, y);
-  }}
+  return (gamma(x) * gamma(y)) / (gamma(x + y));
```

Comment:
This is prone to overflow and hence usually calculations are performed in log space. However, even then one has to be careful that no cancellation errors occur. See e.g. [the explanation and implementation in the boost libraries](https://www.boost.org/doc/libs/1_46_1/libs/math/doc/sf_and_dist/html/math_toolkit/special/sf_beta/beta_function.html#math_toolkit.special.sf_beta.beta_function.implementation) or the [Julia implementations](https://github.com/JuliaMath/SpecialFunctions.jl/blob/9116108ba52d103a05d47d9af1ca6b959a536c30/src/gamma.jl#L769-L803).

Reply:
Thanks for this! I'll have a look at changing this implementation to follow this.